### PR TITLE
Fix one share edge case

### LINKED ISSUE #14: Cannot repair codes with 2 chunks in them
```
goos: linux
goarch: amd64
pkg: github.com/celestiaorg/rsmt2d
cpu: 11th Gen Intel(R) Core(TM) i9-11900H @ 2.50GHz
BenchmarkRepair/RSGF8_1x1x256_ODS-16         	   98379	     12215 ns/op
BenchmarkRepair/LeopardFF8_1x1x256_ODS-16    	panic: runtime error: index out of range [1] with length 1

goroutine 53 [running]:
github.com/celestiaorg/go-leopard.Decode({0xc000108990?, 0x1, 0x0?}, {0xc0001089a8?, 0x1, 0xc00006dcc8?})
	/home/mus/go/pkg/mod/github.com/celestiaorg/go-leopard@v0.1.0/wrapper.go:171 +0x14a
github.com/celestiaorg/rsmt2d.leoRSFF8Codec.Decode({}, {0xc000108990?, 0x30?, 0x0?})
	/home/mus/Code/rsmt2d/leopard.go:26 +0x54
github.com/celestiaorg/rsmt2d.(*ExtendedDataSquare).rebuildShares(0xc0003c8000, 0x0, {0xc000108990?, 0x2, 0x2})
	/home/mus/Code/rsmt2d/extendeddatacrossword.go:234 +0x58
github.com/celestiaorg/rsmt2d.(*ExtendedDataSquare).solveCrosswordRow(0xc0003c8000, 0x0, {0xc0004216e0, 0x2, 0x2}, {0xc000421710, 0x2, 0x2})
	/home/mus/Code/rsmt2d/extendeddatacrossword.go:131 +0x1d4
github.com/celestiaorg/rsmt2d.(*ExtendedDataSquare).solveCrossword(0xc0003c8000, {0xc0004216e0, 0x2, 0x2}, {0xc000421710, 0x2, 0x2})
	/home/mus/Code/rsmt2d/extendeddatacrossword.go:83 +0xd6
github.com/celestiaorg/rsmt2d.(*ExtendedDataSquare).Repair(0xc00014ad80?, {0xc0004216e0, 0x2, 0x2}, {0xc000421710, 0x2, 0x2})
	/home/mus/Code/rsmt2d/extendeddatacrossword.go:66 +0x7a
github.com/celestiaorg/rsmt2d.BenchmarkRepair.func1(0xc00014ad80)
	/home/mus/Code/rsmt2d/extendeddatacrossword_test.go:209 +0x21c
testing.(*B).runN(0xc00014ad80, 0x1)
	/usr/local/go/src/testing/benchmark.go:193 +0x102
testing.(*B).run1.func1()
	/usr/local/go/src/testing/benchmark.go:233 +0x59
created by testing.(*B).run1
	/usr/local/go/src/testing/benchmark.go:226 +0x9c
exit status 2
FAIL	github.com/celestiaorg/rsmt2d	2.837s

```

### PR DESCRIPTION
**Pull request by liamsi**
closes #14

### CODE REVIEW

**Comment by musalbas** [comment]
Can you summarise the cause of the bug, out of curiosity?

File: wrapper.go [hunk 1]
```diff
@@ -168,7 +168,11 @@ func Decode(orig, recovery [][]byte) (decoded [][]byte, err error) {
 	}
 	for i := 0; i < len(recovery); i++ {
 		if recovery[i] != nil {
-			decoded[i+len(orig)] = recovery[i]
+			offset := 0 // special case: 1 share
+			if len(orig) > 1 {
+				offset = len(orig)
+			}
+			decoded[i+offset] = recovery[i]
```
Comment (adlerjohn) [inline wrapper.go:175]: This doesn't look right. If there's one original share and one parity share, then line 175 will assign `decoded[0] = recovery[0]` then terminate. The value here may be correct (polynomial interpolation of a single point is a single point?), but it doesn't address the root cause. `decoded` should have length `2`, not length `1`, and therefore does not address #14.

Comment (liamsi) [reply wrapper.go:175]: > decoded should have length 2, not length 1, and therefore this change does not fix the root cause of https://github.com/celestiaorg/go-leopard/issues/14 even if it hides the symptoms.

hmm, maybe but that is not what the underlying leopard library returns in that case:

https://github.com/catid/leopard/blob/22ddc7804998d31c8f1a2617ee720e063b1fa6cd/leopard.cpp#L207-L208

Comment (liamsi) [reply wrapper.go:175]: Let me dig into this again.

**Comment by liamsi** [comment]
> Can you summarise the cause of the bug, out of curiosity?

As described above, the case with only one share, returns only one share on [`Recover`](https://github.com/celestiaorg/go-leopard/blob/0ff9a9ee85cb13873aa19aeeb369f87ec4444d58/wrapper.go#L101) as well. (due to [this](https://github.com/catid/leopard/blob/22ddc7804998d31c8f1a2617ee720e063b1fa6cd/leopard.cpp#L207-L208) and [this](https://github.com/catid/leopard/blob/22ddc7804998d31c8f1a2617ee720e063b1fa6cd/leopard.cpp#L278-L283)). For all other cases it's 2*origDataLen. (2 shares -> 4 shares on recover and so on). 
I simply quickly made sure that this works with one share in all cases (i.e. lose one share/lose none) and did not want to change the underlying C code (which a. I assumed handles this case correctly and b. would have need to change to achieve what @adlerjohn is describing above:
> decoded should have length 2, not length 1

File: wrapper.go (hunk 1 above)
Comment (liamsi) [reply wrapper.go:175]: > (polynomial interpolation of a single point is a single point, the point itself??)

Also, yes. See what happens on encoding if len(original data) == 1:
https://github.com/catid/leopard/blob/22ddc7804998d31c8f1a2617ee720e063b1fa6cd/leopard.cpp#L143-L149

**Review by adlerjohn — COMMENTED** [review]
Can you document in the code a summary of _why_ the 1-share case is special? Then appooorve.

**Review by adlerjohn — APPROVED** [review]

**Pull request merged**